### ISSUE TITLE
jxl_emcc.cc is not endian-neutral

### ISSUE THREAD

**Issue by thebombzen** [libjxl/libjxl#518]
**Describe the bug**
The code at `tools/jxl_emcc.cc` on line 36 assumes little-endian hardware, and will fail on big-endian hardware.
https://github.com/libjxl/libjxl/blob/aa1ce6c8d172baa6805ea688184cdedb40ec4c06/tools/jxl_emcc.cc#L36

```cpp
size_t compressed_size = compressed.size();
uint8_t* result = reinterpret_cast<uint8_t*>(malloc(compressed_size + 4));
uint32_t* meta = reinterpret_cast<uint32_t*>(result);
meta[0] = compressed_size;
```

**To Reproduce**
Compile this for big-endian hardware (if you have any lying around, for some reason) and the `uint32_t` assigned here will be byte-flipped.

**Expected behavior**
The reference encoder should be endian-agnostic, or at least provide a compile-time check.

**Comment by veluca93** [comment]
IIRC the file is meant to be used with emscripten, which I *believe* is always little endian... is that incorrect?

**Comment by thebombzen** [comment]
WebAssembly is 32-bit and little-endian by current spec, but assuming that a file will be used by a particular compiler against a particular architecture, even if that's the original reason it will written, is not future-proof. This function and the one below it also drop a `size_t` into a `uint32_t`-sized slot, which is another problem.

**Comment by veluca93** [comment]
Fair enough. @eustas thoughts?

**Comment by mo271** [comment]
in part fixed by #1411

**Comment by szabadka** [comment]
After the rewrite using the API, this file does not have this problem any more.
